src/metaxy/models/feature_spec.py

```diff
@@ -164,6 +165,19 @@ def table_name(self) -> str:
         return self.key.table_name
 
 
+def _freeze_metadata(value: Any) -> Any:
```

Comment:
I don't think we should be doing that, Pydantic has `model_config = pydantic.Config(frozen=True)`.  We have it on a helper class in `models/bases.py` or smth like this